[NO QA] feat: Step 3 UI

MrMuzyk opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
This PR introduces UI for step 3 in global reimbursements flow.

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/50899
PROPOSAL:


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

Same as QA steps
- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- Have debug mode enabled
- Go to workspace with currency set to CAD, EUR, GBP or AUD
- Connect Bank Account
- Go to step 3
- UI gathering all required data for this step should be visible
- [x] Verify that no errors appear in the JS console

Note:
AddressFormFields component was updated. While everything should remain exactly the same following places were affected by this:
- Address Form in EnablePayments
- Address Form in BusinessInfo sub step in Reimbursements for USD accounts
- Address Form in PersonalInfo sub step in Reimbursements for USD accounts
- Address Form in BeneficialOwnerInfo sub step in Reimbursements for USD accounts

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/90ce9a58-1a87-4b89-b2a4-7ee094a061e5


</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/604142bd-2af2-4d57-998a-0de2af98beec


</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/818ba863-4369-4869-9e82-4e2671d2403b


</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/a814c78a-3620-4a42-ad87-f1704d5b4025


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/46905375-1dc8-415b-8129-f431825c5c51


</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/f0df1dd2-9843-49a5-b981-e1dfb66d808e


<!-- add screenshots or videos here -->

</details>

MrMuzyk committed:
feat: step 3

MrMuzyk committed:
Merge branch 'main' of github.com:Expensify/App into feat/step-3-ui

MrMuzyk committed:
Merge branch 'main' of github.com:Expensify/App into feat/step-3-ui

MrMuzyk committed:
Merge branch 'main' of github.com:Expensify/App into feat/step-3-ui

MrMuzyk committed:
feat: step 3 pt 2

MrMuzyk committed:
feat: remove mocked provinces and add comment to remaining mock

MrMuzyk committed:
Merge branch 'main' of github.com:Expensify/App into feat/step-3-ui

MrMuzyk committed:
fix: prettier

MrMuzyk committed:
fix: linter, ts

MrMuzyk committed:
Merge branch 'main' of github.com:Expensify/App into feat/step-3-ui

MrMuzyk committed:
fix: simplify PushRowWithModal

MrMuzyk committed:
fix: polishing fixes

MrMuzyk committed:
fix: validation fixes

MrMuzyk committed:
fix: replace substring

MrMuzyk committed:
Merge branch 'main' of github.com:Expensify/App into feat/step-3-ui

MrMuzyk committed:
fix: AddressBusiness

MrMuzyk committed:
fix: change resetting so autocomplete works

MrMuzyk commented on src/languages/es.ts:
```diff
@@ -336,6 +337,7 @@ const translations = {
             pleaseSelectOne: 'Seleccione una de las opciones.',
             invalidRateError: 'Por favor, introduce una tarifa válida.',
             lowRateError: 'La tarifa debe ser mayor que 0.',
+            email: 'Please enter a valid email address.',
```
I'm waiting for final translations - https://expensify.slack.com/archives/C01GTK53T8Q/p1730378501391549

MrMuzyk committed:
fix: move lists outside of component

hungvu193 commented:
I'll review this weekend!

hungvu193 commented:
I think we missed some regexes for zipcode:

https://github.com/user-attachments/assets/ee54e1de-bb4f-4a6e-8d04-80ea5fef7fef


Here's the old regex for zipcode:
https://github.com/Expensify/App/blob/bef062b4caa7f665159dc107911e708031e648c4/src/components/AddressForm.tsx#L121-L131

hungvu193 commented:
Also please add the validation for invalid phone number:


https://github.com/user-attachments/assets/c3abc22d-1828-470d-b36d-41e9cb1ac3e6

hungvu193 commented:
If we choose other country that don't have state option visible, we should also remove the validation for it:

https://github.com/user-attachments/assets/a000422a-6a91-44ef-ae29-5f307d6f6ffb

MrMuzyk commented:
> I think we missed some regexes for zipcode:

This seems to me like a regex for US zipcodes only (I've tried to enter mine which is Polish and it wasn't passing). I'll try to find a universal one.

MrMuzyk committed:
fix: cr fixes and translations

MrMuzyk committed:
fix: typings

MrMuzyk committed:
Merge branch 'main' of github.com:Expensify/App into feat/step-3-ui

MrMuzyk commented:
@hungvu193 I've fixed all of the comments and got approval on translations. Everything should work correctly now. Validation for zipCode is rather loose because of how diverse these codes can be.

I might have to look into typings in FormProvider in the new method I've added but Im not 100% sure.

hungvu193 commented:
I got this crashed while entering VBBA flow, I don't think it comes from our PR, so can you please merge main again? ty

https://github.com/user-attachments/assets/64bcc885-1f80-4c04-90d9-eec372de933c

hungvu193 commented:
Here's the offending PR btw, looks like it's still in progress (https://github.com/Expensify/App/pull/51981)

MrMuzyk committed:
Merge branch 'main' of github.com:Expensify/App into feat/step-3-ui

MrMuzyk committed:
fix: fix crash

MrMuzyk committed:
fix: minor correction

MrMuzyk commented:
@hungvu193 I've applied the fix from the offending PR in here so you can finish review process

hungvu193 commented:
Thanks I'll continue the testing 👍

hungvu193 commented on src/pages/ReimbursementAccount/NonUSD/BusinessInfo/substeps/ContactInformation.tsx:
```diff
@@ -0,0 +1,94 @@
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.REIMBURSEMENT_ACCOUNT_FORM>): FormInputErrors<typeof ONYXKEYS.FORMS.REIMBURSEMENT_ACCOUNT_FORM> => {
+            const errors = ValidationUtils.getFieldRequiredErrors(values, STEP_FIELDS);
+
+            if (values[BUSINESS_CONTACT_NUMBER] && !ValidationUtils.isValidE164Phone(values[BUSINESS_CONTACT_NUMBER])) {
```
I think this validation is not enough.
Should be something like this:
```javascript

            const phoneNumber = LoginUtils.appendCountryCode(values[INPUT_IDS.PHONE_NUMBER]);
            const parsedPhoneNumber = PhoneNumberUtils.parsePhoneNumber(values[BUSINESS_CONTACT_NUMBER]);
            if (!parsedPhoneNumber.possible || !Str.isValidE164Phone(phoneNumber.slice(0))) {
                errors[BUSINESS_CONTACT_NUMBER] = translate('bankAccount.error.phoneNumber');
            }
            ret
```

MrMuzyk replied:
I wasn't aware this exists but it makes sense - changing it

hungvu193 commented:
> If we choose other country that don't have state option visible, we should also remove the validation for it:

@MrMuzyk  Now, we have another bug, `State` field sometimes appears with non-us country.

https://github.com/user-attachments/assets/60541ffc-3f8f-4c83-88d2-d249dd7884fb

MrMuzyk committed:
fix: Improve phone number validation

MrMuzyk commented:
>@MrMuzyk Now, we have another bug, State field sometimes appears with non-us country.
Sometimes, State field won't appear with US country.

It's probably caused by the fact that when user comes back to any of the sub steps from the confirmation screen we don't want to save anything he changed unless he confirms it. Looking for a fix now

MrMuzyk committed:
fix: fix state displaying and validation

MrMuzyk committed:
Merge branch 'main' of github.com:Expensify/App into feat/step-3-ui

MrMuzyk commented:
@hungvu193 I've fixed known issues and found a few more edge cases when it came to autocomplete and editing from confirmation screen. You can check again

hungvu193 commented:
Thank you. I'll check again in a while 💯

hungvu193 commented:
I still can reproduce the issue, here are the steps:
1. Go to Address page.
2. Choose US, press Submit without filling in the state.
3. Choose the state.
4. Choose another country.

https://github.com/user-attachments/assets/88706db4-68e5-4252-98a6-c43ba04cc6ac

MrMuzyk committed:
fix: reset state errors whenever stepField change

MrMuzyk committed:
Merge branch 'main' of github.com:Expensify/App into feat/step-3-ui

hungvu193 commented on src/libs/ValidationUtils.ts:
```diff
@@ -503,6 +503,33 @@ function isValidSubscriptionSize(subscriptionSize: string): boolean {
+    const phoneNumberWithCountryCode = LoginUtils.appendCountryCode(phoneNumber);
+    const parsedPhoneNumber = parsePhoneNumber(phoneNumberWithCountryCode);
+
+    return Str.isValidE164Phone(parsedPhoneNumber.number?.e164 ?? '');
```
```suggestion
    return parsedPhoneNumber.possible && Str.isValidE164Phone(parsedPhoneNumber.number?.e164 ?? '');
```

hungvu193 replied:
Let's use add `parsedPhoneNumber.possible` into the condition. `isValidE164Phone` uses [this regex](https://github.com/Expensify/expensify-common/blob/2f96348d48fb9657b95aae12567e6846e199da88/lib/CONST.ts#L629), which is not enough.

<img width="1020" alt="Screenshot 2024-11-06 at 17 36 14" src="https://github.com/user-attachments/assets/16ef9546-860a-44d7-92e3-1bef7f82ee79">

MrMuzyk replied:
Updated

hungvu193 replied:
All good! I'll work on the checklist!

hungvu193 reviewed (commented):
All good so far! One comment left.

MrMuzyk committed:
fix: tighten phone number validation

hungvu193 commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos 
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->


https://github.com/user-attachments/assets/9f49b42e-c93f-4964-8c57-9dc45dcbe319



</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/707404fb-23b7-4d66-945a-0356da87034a

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/9e2fa34e-82dc-4aa3-8fad-75d55265089c

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/53f774e5-5134-4583-9aa5-54ad8a759e89

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->


https://github.com/user-attachments/assets/3d2e627a-c430-470d-83d8-d9403ec03231



</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/9ee3e34e-2bae-42fe-8203-2cbeea8d6c7f

</details>

hungvu193 reviewed (approved):
Tests well!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

IuliiaHerets commented:
@MrMuzyk @hungvu193 @madmax330, the QA team sees a request to update to USD when connecting to a bank account, even with currencies mentioned in the PR. How can we connect BA then? Is it an internal PR? 

https://github.com/user-attachments/assets/371a2b83-177a-4ba7-8345-186eb171a1fd

IuliiaHerets commented:
@MrMuzyk Flow doesn't work on our side with enabling debug mode via the troubleshooting menu. We still need to change the currency to USD to continue

https://github.com/user-attachments/assets/21f3c10a-c689-4e40-9520-7058701a9ffb

MrMuzyk commented:
Ok. I know where the issue is.

https://github.com/Expensify/App/blob/main/src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx#L234
In this line this check has to get updated to `if (isDevelopment || isDebugModeEnabled) {`

Is this a blocker or can I add that in the next PR im preparing as part of this new global reimbursements flow?

hungvu193 commented:
Users can't access this feature anyway, so I don't think it's blocker, let update the title of this PR with No QA prefix and make sure we will add above condition to our next PR

Pujan92 commented on src/components/Form/FormProvider.tsx:
```diff
@@ -244,9 +245,20 @@ function FormProvider(
+        (inputID: keyof Form) => {
+            const newErrors = {...errors};
+            delete newErrors[inputID];
+            FormActions.setErrors(formID, newErrors as Errors);
```
local validation errors do not belong in the Onyx server-error channel, which caused an issue https://github.com/Expensify/App/issues/94417
